src/components/legend/draw.js

```diff
@@ -460,17 +467,25 @@ function setupTraceToggle(g, gd) {
     });
 }
 
+function textLayout(s, g, gd) {
+    svgTextUtils.convertToTspans(s, gd, function() {
+        computeTextDimensions(g, gd);
+    });
+}
+
 function computeTextDimensions(g, gd) {
     var legendItem = g.data()[0][0];
-
-    if(!legendItem.trace.showlegend) {
+    if(legendItem && !legendItem.trace.showlegend) {
         g.remove();
         return;
     }
 
     var mathjaxGroup = g.select('g[class*=math-group]');
     var mathjaxNode = mathjaxGroup.node();
-    var opts = gd._fullLayout.legend;
+    var bw = gd._fullLayout.legend.borderwidth;
+    var opts = legendItem ?
+        gd._fullLayout.legend :
+        gd._fullLayout.legend.title;
```

Comment:
I'm not a big fan of making `computeTextDimensions` work for the legend items AND the legend title.

What if you revert `computeTextDimensions` to the way it was and added a similar (but different enough) `computeTitleDimensions`?

Reply:
I don't like the complexities you brought into `computeTextDimensions`, but I don't like them enough to block this PR.

Reply:
... so don't be surprise I change the `computeTextDimensions` pattern in a future legend PR of mine.

Reply:
Sure. Please go for it.